cassandra/io/twistedreactor.py

```diff
@@ -133,6 +135,25 @@ def _cleanup(self):
                             "Cluster.shutdown() to avoid this.")
             log.debug("Event loop thread was joined")
 
+    def add_timer(self, timer):
+        self._timers.add_timer(timer)
+        reactor.callFromThread(self._schedule_timeout, timer.end)
```

Comment:
Okay, that makes sense to me and is what I expected.  I would just add a comment to make that clear.